activity/worker_test.go

```diff
@@ -0,0 +1,340 @@
+	return w.activity(ctx, "two", &Input1{Data: "two"})
+}
+
+func (w *WorkerFSM) activity(ctx *fsm.FSMContext, name string, input interface{}) swf.Decision {
+	var serialized *string
+	if input != nil {
+		serialized = S(ctx.Serialize(input))
+	}
+	return swf.Decision{
+		DecisionType: S(swf.DecisionTypeScheduleActivityTask),
+		ScheduleActivityTaskDecisionAttributes: &swf.ScheduleActivityTaskDecisionAttributes{
+			ActivityID:             S(name),
+			ActivityType:           &swf.ActivityType{Name: S(name), Version: S(name)},
+			Input:                  serialized,
+			HeartbeatTimeout:       S("30"),
+			ScheduleToStartTimeout: S("NONE"),
+			ScheduleToCloseTimeout: S("NONE"),
+			StartToCloseTimeout:    S("NONE"),
+			TaskList:               &swf.TaskList{Name: S("aTaskListSharedBetweenTaskOneAndTwo")},
+		},
+	}
+}
+
+func TestTypedActivityWorker(t *testing.T) {
+	if os.Getenv("AWS_ACCESS_KEY_ID") == "" || os.Getenv("AWS_SECRET_ACCESS_KEY") == "" {
+		log.Printf("WARNING: NO AWS CREDS SPECIFIED, SKIPPING MIGRATIONS TEST")
+		return
+	}
+
+	creds, _ := aws.EnvCreds()
+	client := swf.New(creds, "us-east-1", nil)
+
+	domain := "worker-test-domain"
+
+	req := swf.RegisterDomainInput{
+		Name:                                   aws.String(domain),
+		Description:                            aws.String("test domain"),
+		WorkflowExecutionRetentionPeriodInDays: aws.String("30"),
+	}
+
+	d := migrator.DomainMigrator{
+		RegisteredDomains: []swf.RegisterDomainInput{req},
+		Client:            client,
+	}
+
+	d.Migrate()
+
+	workflow := "worker-test-workflow"
+	version := "worker-test-workflow-version"
+
+	wreq := swf.RegisterWorkflowTypeInput{
+		Name:            &workflow,
+		Description:     aws.String("test workflow migration"),
+		Version:         &version,
+		Domain:          aws.String(domain),
+		DefaultTaskList: &swf.TaskList{Name: S("worker-fsm")},
+	}
+
+	w := migrator.WorkflowTypeMigrator{
+		RegisteredWorkflowTypes: []swf.RegisterWorkflowTypeInput{wreq},
+		Client:                  client,
+	}
+
+	w.Migrate()
+
+	one := swf.RegisterActivityTypeInput{
+		Name:        aws.String("one"),
+		Description: aws.String("worker test activity 1"),
+		Version:     aws.String("one"),
+		Domain:      aws.String(domain),
+	}
+
+	two := swf.RegisterActivityTypeInput{
+		Name:        aws.String("two"),
+		Description: aws.String("worker test activity 2"),
+		Version:     aws.String("two"),
+		Domain:      aws.String(domain),
+	}
+
+	a := migrator.ActivityTypeMigrator{
+		RegisteredActivityTypes: []swf.RegisterActivityTypeInput{one, two},
+		Client:                  client,
+	}
+
+	a.Migrate()
+
+	taskList := "aTaskListSharedBetweenTaskOneAndTwo"
+
+	worker := &ActivityWorker{
+		Domain:     domain,
+		Serializer: fsm.JSONStateSerializer{},
+		TaskList:   taskList,
+		SWF:        client,
+		Identity:   "test-activity-worker",
+	}
+
+    //This is where the actual worker config code is
```

Comment:
Here is how we wire up the worker.  the machinery takes care of all the serialization/deserialization, calling done/fail, etc.